src/pages/dashboard/seller/Sales/UpdateOrderModal.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../../../redux/hooks/hooks';
+import {
+	closeModal,
+	getSales,
+	setDeliveryDate,
+	setStatus,
+} from '../../../../redux/features/Sales/AllSaleSlice';
+import { updateOrder } from '../../../../redux/features/Sales/UpdateorderSlice';
+import useToast from '../../../../hooks/useToast';
+import { DynamicData } from '../../../../@types/DynamicData';
+
+const UpdateOrderModal: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const { selectedSaleId, deliveryDate, status, productName, productImage } =
+		useAppSelector((state) => state.sales);
+
+	const { showErrorMessage, showSuccessMessage } = useToast();
+
+	const handleUpdateOrder = async (event: React.FormEvent) => {
+		event.preventDefault();
+		try {
+			const resultAction = await dispatch(
+				updateOrder({
+					id: selectedSaleId ?? '',
+					status: status,
+					deliveryDate: deliveryDate ?? '',
+				}),
+			).unwrap();
+
+			if (resultAction) {
+				dispatch(closeModal());
+				showSuccessMessage(resultAction.message);
+				dispatch(getSales()).unwrap();
+			}
+		} catch (e) {
+			const error = e as DynamicData;
+			let errorMessage = 'An error occurred';
+			if (
+				error.response &&
+				error.response.data &&
+				error.response.data.message
+			) {
+				errorMessage = error.response.data.message;
+			} else if (error.message) {
+				errorMessage = error.message;
+			}
+
+			showErrorMessage(errorMessage);
+		}
+	};
+
+	const handleOverlayClick = (e: React.MouseEvent) => {
+		if (e.target === e.currentTarget) {
+			dispatch(closeModal());
+		}
+	};
+
+	return (
+		<div
+			className="fixed top-0 bottom-0 left-0 right-0 flex items-center justify-center bg-modalOverlay z-50"
+			onClick={handleOverlayClick}
+			data-testid="modal-overlay"
+		>
+			<div className="relative bg-neutral-white p-12 w-2/5 rounded-md min-w-fit">
+				<button
+					onClick={() => dispatch(closeModal())}
+					className="absolute top-2 right-2 bg-action-error rounded-full py-1 px-3 text-lg text-neutral-white"
+				>
+					X
+				</button>
+				<div className="mb-4 text-center">
+					<img
+						src={productImage}
+						alt={productName}
+						className="w-40 h-32 mx-auto rounded-md"
+					/>
+					<h2 className="text-lg font-semibold">{productName}</h2>
+					<p className="text-sm text-gray-600">Current Status: {status}</p>
+				</div>
+				<form onSubmit={handleUpdateOrder} className="justify-between">
+					<div className="mb-2">
+						<label htmlFor="status" className="block text-sm font-medium">
+							Status
+						</label>
+						<select
+							id="status"
+							value={status}
+							onChange={(e) => dispatch(setStatus(e.target.value))}
+							className="mt-1 block w-full pl-3 pr-10 py-2 border-2 text-base mobile:text-sm rounded-md"
+						>
+							<option value="delivered">Delivered</option>
```

Comment:
Why only two statuses while we have three?

Reply:
By default, the status is pending right after the order is made but when we are updating the order what is only allowed is to cancel it or deliver it.